config/routes.rb

```diff
@@ -107,6 +98,8 @@
         resource :moderator, only: %i[create destroy], module: "tags"
       end
       resources :users, only: %i[index show edit update] do
+        get "email_message/:id", to: "users#email_message"
```

Comment:
Maybe I'm missing something, but how about a normal resource instead? 

```ruby
resources :email_message, only: :show
```

A separate controller seems cleaner than putting more responsibility into `Admin::UsersController` and we'd get a path helper that you could use instead of manually constructing the URL like this:

```ruby
<a href="/admin/users/<%= @user.id %>/email_message/<%= email.id %>"
```

Reply:
That would require adding a controller and not just an action to an existing one. An alternative is to use:

```ruby
#  (message is a bit redundant?)
get "email_message/:id", to: "users#email_message", as: :users_email_message
```

which will also build a useful helper to avoid harcoding the url

Reply:
I'm very much in favor of more controllers with only restful actions instead of adding arbitrary actions to existing controllers.

Reply:
Agreed, it also plays in the wider discussion in coming up with an "internal best practices" kind of document

Reply:
Great feedback, thanks @citizen428  and @rhymes! I'll get this restructured soon